cmd/gqltool/main.go

```diff
@@ -40,10 +40,16 @@ func main() {
 }
 
 func getDefaultConfigHome() string {
-	// TODO(adamb): switch to os.UserConfigDir()
-	dir, err := os.UserHomeDir()
+	dir, err := os.UserConfigDir()
 	if err != nil {
 		panic(err)
 	}
-	return filepath.Join(dir, ".config", "stateful")
+	_, fErr := os.Stat(dir)
+	if os.IsNotExist(fErr) {
+		mkdErr := os.MkdirAll(dir, os.ModePerm)
+		if mkdErr != nil {
+			dir = os.TempDir()
+		}
+	}
+	return filepath.Join(dir, "stateful")
```

Comment:
Do you think we could eliminate the code duplication here by making the `internal` function public and importing it here? 

If we were ever gonna unify the two implementations, now would be a good time to do that for sure.

Reply:
This was implemented